packages/flutter_tools/test/general.shard/ios/ios_deploy_test.dart

```diff
@@ -98,6 +98,26 @@ void main () {
           '', '']);
       });
 
+      testWithoutContext('app exit', () async {
+        final FakeProcessManager processManager = FakeProcessManager.list(<FakeCommand>[
+          const FakeCommand(
+            command: <String>['ios-deploy'],
+            stdout: '(lldb)     run\r\nsuccess\r\nLog on attach\r\nProcess 100 exited with status = 0\r\nLog after process exit',
+          ),
+        ]);
+        final IOSDeployDebugger iosDeployDebugger = IOSDeployDebugger.test(
+          processManager: processManager,
+          logger: logger,
+        );
+        final List<String> receivedLogLines = <String>[];
+        final Stream<String> logLines = iosDeployDebugger.logLines
+          ..listen(receivedLogLines.add);
+
+        expect(await iosDeployDebugger.launchAndAttach(), isTrue);
+        await logLines.toList();
```

Comment:
you could also `await logLines.last`

Reply:
Hm, it doesn't like that:
```
dart:async                                          Stream.last
test/general.shard/ios/ios_deploy_test.dart 117:24  main.<fn>.<fn>.<fn>
===== asynchronous gap ===========================
dart:async                                          _asyncThenWrapperHelper
test/general.shard/ios/ios_deploy_test.dart         main.<fn>.<fn>.<fn>
dart:async                                          runZoned
test/src/common.dart 207:14                         testWithoutContext.<fn>
test/src/common.dart 177:18                         test.<fn>

Bad state: No element
```

Reply:
Ahh, because it is empty...

Reply:
toList seems easier than Listen/onDone

Reply:
Getting the streams to work in these tests was 3x more work than the actual change.